client/pages/userAccount.jsx

```diff
@@ -0,0 +1,209 @@
+        if (result[0].length !== 0) {
+          result[0].forEach(element => {
+            const stateCode = element.stateCode;
+            if (defaultStates[stateCode]) {
+              defaultStates[stateCode].visits = element.visits;
+            }
+          });
+          this.renderInforgraphic();
+          this.setState({
+            results: result[0],
+            total: result[1][0].reviews
+          });
+        } else {
+          this.renderInforgraphic();
+          this.setState({
+            results: null,
+            total: 'N/A'
+          });
+        }
+      })
+      .catch(err => console.error(err));
+  }
+
+  renderInforgraphic() {
+    const dataObject = {};
+    for (const key in defaultStates) {
+      const stateName = defaultStates[key].name;
+      const visits = defaultStates[key].visits;
+      dataObject[stateName] = Number(visits);
+    }
+    const color = d3.scaleQuantize()
+      .domain([1, 15])
+      .range(d3.schemeGreens[9]);
+
+    const path = d3.geoPath();
+    const svg = d3.select('#map')
+      .append('svg')
+      .attr('viewBox', '0 0 975 610');
+
+    const toolTip = d3.select('#map')
+      .append('div')
+      .style('position', 'absolute')
+      .attr('class', 'tooltip');
+
+    d3.json('https://cdn.jsdelivr.net/npm/us-atlas@3/states-albers-10m.json')
+      .then(us => {
+        svg.append('g')
+          .selectAll('path')
+          .data(topojson.feature(us, us.objects.states).features)
+          .enter().append('path')
+          .attr('d', path)
+          .attr('class', 'states')
+          .style('fill', d => color(dataObject[d.properties.name]))
+          .style('stroke', '#636363');
+
+        svg.selectAll('path')
+          .on('mouseover', function (event, d) {
+            d3.selectAll('.states')
+              .transition()
+              .duration(200)
+              .style('stroke-width', '1px')
+              .style('opacity', 0.8)
+              .style('box-shadow', '2px 2px 0.5rem black');
+
+            d3.select(this)
+              .transition()
+              .duration(200)
+              .style('opacity', 1)
+              .style('stroke-width', '2px')
+              .style('cursor', 'pointer');
+
+            d3.select('.tooltip')
+              .style('opacity', 1);
+
+          })
+          .on('mouseout', function (event) {
+            d3.selectAll('.states')
+              .transition()
+              .duration(200)
+              .style('opacity', 1)
+              .style('stroke-width', '1px');
+
+            toolTip.style('opacity', 0);
+          })
+          .on('mousemove', function (event, d) {
+            const offsetX = event.layerX > parent.innerWidth * 0.5 ? '-105%' : '5%';
+            const offsetY = event.layerY > parent.innerWidth * 0.5 ? '-105%' : '5%';
+            const offset = `translate(${offsetX}, ${offsetY})`;
+
+            toolTip
+              .html(`<h6 class='open-sans mb-0 mt-2'>${d.properties.name}</h6>
```

Comment:
We should definitely `escape()` this `d.properties.name` string since we don't control it. (xss)